app/Filters/PaymentFilters.php

```diff
@@ -42,6 +42,26 @@ public function filter(string $filter = '') : Builder
         });
     }
 
+    /**
+     * Filters the list based on the transaction reference
+     * Used to sincronize status with external apis
+     *
+     * @param string transaction
+     * @return Builder
+     */
+    public function transaction(string $transaction = '') : Builder
+    {
+	    $table = 'payments';
+        if (strlen($transaction) == 0) {
+            return $this->builder->where(function ($query) use ($table) {
+                $query->whereNull($table.'.transaction_reference')
+                    ->orWhere($table.'.transaction_reference', 'like', '');
```

Comment:
I don't think we should be returning transaction_references with null values, I would suggest returning only those that hit some form of match, also your LIKE statement won't work without prepending or appending % values

Reply:
i use that to realize a chekup after sincronize errors.
if a webhook that place a payment fail to set the reference, with this filter i can find these payments and correct place the values.

Reply:
its not a catchall filter.
only nulls.

Reply:
i have updated the N8N node of InvoiceNinja v5, but i dont know how to make it public.
would be great if you can teach me.